packages/utils/src/ast.ts

```diff
@@ -158,13 +161,31 @@ function breakTokensTemplateLiteral(
     );
 
     return [
-      [...output[0], templateElement({ raw: suffix, cooked: suffix })],
+      [
+        ...output[0],
+        templateElement({ raw: getRawTemplateValue(suffix), cooked: suffix }),
+      ],
       output[1],
     ];
   }
 
   // If there are no matches, simply return the original value.
-  return [[templateElement({ raw: value, cooked: value })], []];
+  return [
+    [templateElement({ raw: getRawTemplateValue(value), cooked: value })],
+    [],
+  ];
+}
+
+/**
+ * Get a raw template literal value from a cooked value. This adds a backslash
+ * before every '`', '\' and '${' characters.
+ *
+ * @see https://github.com/babel/babel/issues/9242#issuecomment-532529613
+ * @param value - The cooked string to get the raw string for.
+ * @returns The value as raw value.
+ */
+function getRawTemplateValue(value: string) {
+  return value.replace(/\\|`|\$\{/gu, '\\$&');
```

Comment:
Is this enough or are there uncovered edge-cases here? The issue you linked calls it a "work in progress"

Reply:
This is the same code as used in [`babel-plugin-syntax-highlight`](https://github.com/jaydenseric/babel-plugin-syntax-highlight/blob/24643c9c41cd0c50f85d03601f095df2b8db53a5/index.js#L111-L113), so I would assume this is fine. But I don't know enough about it to be sure.